### DISCUSSION TITLE
Website fails to deploy after new commit

### DISCUSSION THREAD

**Discussion by scvcrem** [alshedivat/al-folio#2120] [Q&A] [answered]
I have built my website using the al-folio repo a couple of years ago. I have regularly update it until today when Github has reported failure to deploy after a new update. I tried to revert all commits, but the website still fails to deploy. This is the log I get. Any hint is much appreciated! 

Run yes | bash bin/deploy --verbose  --src master --deploy gh-pages
+ shift
+ [[ 4 > 0 ]]
Deploying...
+ key=--src
+ case $key in
+ SRC_BRANCH=master
+ shift
+ shift
+ [[ 2 > 0 ]]
+ key=--deploy
+ case $key in
+ DEPLOY_BRANCH=gh-pages
+ shift
+ shift
+ [[ 0 > 0 ]]
+ set -e
+ echo Deploying...
+ echo 'Source branch: master'
+ echo 'Deploy branch: gh-pages'
+ read -r -p 'Do you want to proceed? [y/N] ' response
Source branch: master
Deploy branch: gh-pages
+ [[ ! y =~ ^([yY][eE][sS]|[yY])+$ ]]
+ git diff-index --quiet HEAD --
++ git ls-files --exclude-standard --others
+ test -z ''
++ git branch
++ grep master
++ tr ' ' '\n'
++ tail -1
+ '[' master ']'
+ git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
++ git branch
++ grep gh-pages
+ '[' ']'
+ git checkout -b gh-pages
Switched to a new branch 'gh-pages'
+ bundle exec jekyll build
bundler: failed to load command: jekyll (/home/runner/work/scvcrem.github.io/scvcrem.github.io/vendor/bundle/ruby/3.0.0/bin/jekyll)
/opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/runtime.rb:304:in `check_for_activated_spec!': You have already activated uri 0.10.1, but your Gemfile requires uri 0.13.0. Since uri is a default gem, you can either remove your dependency on it or try updating to a newer version of bundler that supports uri as a default gem. (Gem::LoadError)
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/runtime.rb:25:in `block in setup'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/spec_set.rb:191:in `each'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/spec_set.rb:191:in `each'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/runtime.rb:24:in `map'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/runtime.rb:24:in `setup'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler.rb:162:in `setup'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/setup.rb:26:in `block in <top (required)>'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/ui/shell.rb:159:in `with_level'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/ui/shell.rb:111:in `silence'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/setup.rb:26:in `<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli/exec.rb:56:in `require_relative'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli/exec.rb:56:in `kernel_load'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli/exec.rb:23:in `run'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli.rb:451:in `exec'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli.rb:34:in `dispatch'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/cli.rb:28:in `start'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/exe/bundle:28:in `block in <top (required)>'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/lib/ruby/gems/3.0.0/gems/bundler-2.5.5/exe/bundle:20:in `<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/bin/bundle:23:in `load'
	from /opt/hostedtoolcache/Ruby/3.0.2/x64/bin/bundle:23:in `<main>'
yes: standard output: Broken pipe
Error: Process completed with exit code 1.

**Comment by george-gca** [comment, marked as answer]
Somehow you still aren't using the latest version of our template. Your repository is way behind the current tip of our `master` branch. As it shows in your `master` branch: `This branch is 75 commits ahead of, 331 commits behind alshedivat/al-folio:master.` If you check your action run, you are using deprecated libraries/commands. This happens because you are using a very old version of al-folio. I suggest [upgrading your code](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#upgrading-from-a-previous-version) to the latest version. More specifically:

Assuming the current directory is `your-repo-name`

```bash
git remote add upstream https://github.com/alshedivat/al-folio.git
git fetch upstream
git rebase v0.11.0
```

Note that probably you will need to do a lot of manual merging. If you find it easier, you could create a copy of your repository, do a fresh install from the template and reapply all your changes. For this I would recommend a tool like [meld](https://meldmerge.org/) or [winmerge](https://winmerge.org/) to check the differences between directories/files. Related issue #2073.

**Reply by scvcrem** [reply]
Thanks George, that's what I feared. I will explore both options.
